rsconnect/actions.py

```diff
@@ -488,767 +265,16 @@ def validate_quarto_engines(inspect):
-            app_mode,
-            environment,
-            extra_files,
-            excludes,
-            image=image,
-            env_management_py=env_management_py,
-            env_management_r=env_management_r,
-        )
-        .deploy_bundle()
-        .save_deployed_info()
-        .emit_task_log()
-    )
-
-
-def deploy_python_api(
-    connect_server: api.TargetableServer,
-    directory: str,
-    extra_files: typing.List[str],
-    excludes: typing.List[str],
-    entry_point: str,
-    new: bool,
-    app_id: int,
-    title: str,
-    python: str,
-    force_generate: bool,
-    log_callback: typing.Callable,
-    image: str = None,
-    env_management_py: bool = None,
-    env_management_r: bool = None,
-) -> typing.Tuple[str, typing.Union[list, None]]:
-    """
-    A function to deploy a Python WSGi API module to Connect.  Depending on the files involved
-    and network latency, this may take a bit of time.
-
-    :param connect_server: the Connect server information.
-    :param directory: the app directory to deploy.
-    :param extra_files: any extra files that should be included in the deploy.
-    :param excludes: a sequence of glob patterns that will exclude matched files.
-    :param entry_point: the module/executable object for the WSGi framework.
-    :param new: a flag to force this as a new deploy. Previous default = False.
-    :param app_id: the ID of an existing application to deploy new files for. Previous default = None.
-    :param title: an optional title for the deploy.  If this is not provided, one will
-    be generated. Previous default = None.
-    :param python: the optional name of a Python executable. Previous default = None.
-    :param force_generate: force generating "requirements.txt" or "environment.yml",
-    even if it already exists. Previous default = False.
-    :param log_callback: the callback to use to write the log to.  If this is None
-    (the default) the lines from the deployment log will be returned as a sequence.
-    If a log callback is provided, then None will be returned for the log lines part
-    of the return tuple. Previous default = None.
-    :param image: the optional docker image to be specified for off-host execution. Default = None.
-    :param env_management_py: False prevents Connect from managing the Python environment for this bundle.
-        The server administrator is responsible for installing packages in the runtime environment. Default = None.
-    :param env_management_r: False prevents Connect from managing the R environment for this bundle.
-        The server administrator is responsible for installing packages in the runtime environment. Default = None.
-    :return: the ultimate URL where the deployed app may be accessed and the sequence
-    of log lines.  The log lines value will be None if a log callback was provided.
-    """
-    return deploy_app(app_mode=AppModes.PYTHON_API, **locals())
-
-
-def deploy_python_fastapi(
```

Comment:
This function is imported by `vetiver-python` and its removal may be why the tests are failing.

Reply:
Ah, that's good to know! This makes me a little concerned, though: I removed many of the functions in `actions.py` because they were not used internally and seemed to essentially be duplicates of functions in `main.py`. Do we need to keep any other of those functions around in case someone external was using them?

Also, in this particular case, I wonder if we can just have `actions.deploy_python_fastapi` be an alias to `main.deploy_python_fastapi`.

Reply:
Currently the only non-CLI clients are `vetiver-python` and `rsconnect-jupyter`.

https://github.com/rstudio/rsconnect-jupyter/blob/0c3d679d9ec799330134aa7710646870ec35e9c9/rsconnect_jupyter/__init__.py#L13C1-L29C45

Reply:
> I wonder if we can just have `actions.deploy_python_fastapi` be an alias to `main.deploy_python_fastapi`

Yes, as long as we're careful to match the type signature that Vetiver is expecting.

Reply:
Hm, there currently isn't a `main.deploy_python_fastapi`; the command line command is generated by the following and isn't saved in a variable anywhere:
```py
generate_deploy_python(app_mode=AppModes.PYTHON_FASTAPI, alias="fastapi", min_version="2021.08.0")
```

Also, the function signature is somewhat different and I don't want to risk breaking things, so for now I'll just bring back this code.